react/withCssHandles.tsx

```diff
@@ -1,20 +1,22 @@
-import React from 'react'
+import React, { ComponentType } from 'react'
+
 import useCssHandles from './useCssHandles'
-import { ComponentType } from 'react'
+import { CssHandlesList, CssHandlesOptions, CssHandles } from './typings'
 
-const withCssHandles = <T extends CssHandlesInput, ComponentProps>(
+const withCssHandles = <T extends CssHandlesList, ComponentProps>(
   handles: T,
-  options?: CssHandlesOptions
+  options?: CssHandlesOptions<T>
 ) => (
   Component: ComponentType<ComponentProps & { cssHandles: CssHandles<T> }>
 ) => {
   const EnhancedComponent = (props: ComponentProps) => {
-    const cssHandles = useCssHandles(handles, options)
+    const { handles: cssHandles } = useCssHandles(handles, options)
```

Comment:
Shouldn't we pass down the whole object? Since this is just a substitute for `useCssHandles`, I think it makes sense to pass everything to `<Component>`, so that it has all the features from `useCssHandles`

Reply:
It makes sense!